sdk/storage/azure-storage-blobs/test/block_blob_client_test.cpp

```diff
@@ -358,8 +358,12 @@ namespace Azure { namespace Storage { namespace Test {
       downloadBuffer.resize(static_cast<std::size_t>(downloadSize), '\x00');
       Blobs::DownloadBlobToOptions options;
       options.Concurrency = concurrency;
-      options.Offset = offset;
-      options.Length = length;
+      if (offset.HasValue() || length.HasValue())
+      {
+        options.Range = Core::Http::Range();
+        options.Range.GetValue().Offset = offset.GetValue();
```

Comment:
nit: `GetValue()` + setting the value is not intuitive. It makes the code a bit more difficult to understand. Also, `(offset.HasValue() || length.HasValue())` does not gurantee that `offset.GetValue()` won't crash.
How about `options.Range->Offset = offset;`?

Reply:
> GetValue() + setting the value is not intuitive. It makes the code a bit more difficult to understand.

There's no way to get around.

> (offset.HasValue() || length.HasValue()) does not gurantee that offset.GetValue() won't crash.

When `length` has value, `offset` always has value. This is just test code, we don't have to be that strict.